core/node/proof_data_handler/src/lib.rs

```diff
@@ -88,17 +96,23 @@ fn create_proof_processing_router(
 
     if config.tee_support {
         let get_tee_proof_gen_processor =
-            TeeRequestProcessor::new(blob_store, connection_pool, config.clone());
+            TeeRequestProcessor::new(blob_store, connection_pool, config.clone(), l2_chain_id);
         let submit_tee_proof_processor = get_tee_proof_gen_processor.clone();
         let register_tee_attestation_processor = get_tee_proof_gen_processor.clone();
 
         router = router.route(
             "/tee/proof_inputs",
             post(
                 move |payload: Json<TeeProofGenerationDataRequest>| async move {
-                    get_tee_proof_gen_processor
+                    let result: Result<Json<zksync_prover_interface::api::TeeProofGenerationDataResponse>, errors::RequestProcessorError> = get_tee_proof_gen_processor
                         .get_proof_generation_data(payload)
```

Comment:
Off-topic (?): The design of `get_proof_generation_data()` looks really weird; it mixes data access and response representation layers which should ideally be cleanly separated. Representing responses in the responsibility of the API server, i.e. *this* code.

Reply:
@pbeza ?

Reply:
Yeah, no need to rework it rn, that's why I've marked discussion as off-topic. IMO, this firmly falls into the category of tech debt (i.e., we'd want to eventually refactor this logic unless it would be removed shortly).